File: backend/components/metiers/administration_publique/utils/plugins/samples/pluginManager.test.js
// pluginManager.test.js
// Test unitaire avancé pour pluginManager (JS)
const pluginManager = require('../core/pluginManager');

describe('pluginManager JS', () => {
  it('enregistre et exécute un plugin', () => {
    pluginManager.register('test', () => 'ok');
    expect(pluginManager.run('test')).toBe('ok');
  });
});

File: backend/components/metiers/administration_publique/utils/plugins/samples/pluginManager.test.py
# pluginManager.test.py
"""Test unitaire avancé pour pluginManager (Python)"""
from ..core.pluginManager import PluginManager

def test_plugin_manager():
    pm = PluginManager()
    pm.register('test', lambda: 'ok')
    assert pm.run('test') == 'ok'
